parsec/core/gui/notification_center_widget.py

```diff
@@ -14,52 +15,53 @@
 
 
 class ItemDelegate(QStyledItemDelegate):
-    def paint(self, painter, option, index):
+    def paint(self, painter: QPainter, option: QStyleOptionViewItem, index: QModelIndex) -> None:
         view_option = QStyleOptionViewItem(option)
         view_option.decorationAlignment |= Qt.AlignHCenter
         # Even though we told Qt that we don't want any selection on our
         # list, it still adds a blue rectangle on focused items.
         # So we just get rid of focus.
         if option.state & QStyle.State_HasFocus:
-            view_option.state &= ~QStyle.State_HasFocus
+            view_option.state &= cast(QStyle.State, ~QStyle.State_HasFocus)
         super().paint(painter, view_option, index)
 
 
 class NotificationCenterWidget(QWidget, Ui_NotificationCenterWidget):
     close_requested = pyqtSignal()
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
         self.setupUi(self)
         self.button_close.clicked.connect(self.close_requested.emit)
 
-    def paintEvent(self, _):
+    def paintEvent(self, _: QPaintEvent) -> None:
         opt = QStyleOption()
         opt.initFrom(self)
         p = QPainter(self)
         self.style().drawPrimitive(QStyle.PE_Widget, opt, p, self)
 
-    def close_notification(self, notif):
+    def close_notification(self, notif: QWidget) -> None:
         self.widget_layout.layout().removeWidget(notif)
         notif.hide()
-        notif.setParent(None)
+        notif.setParent(None)  # type: ignore[call-overload]
 
-    def clear(self):
+    def clear(self) -> None:
         while self.widget_layout.layout().count() > 1:
             item = self.widget_layout.layout().takeAt(0)
             if item.widget():
                 item.widget().hide()
                 item.widget().setParent(None)
 
-    def add_notification(self, notif_type, msg):
-        widget = None
+    def add_notification(self, notif_type: str, msg: str) -> None:
         if notif_type == "ERROR":
-            widget = ErrorNotificationWidget()
+            widget: ErrorNotificationWidget | WarningNotificationWidget | InfoNotificationWidget = (
```

Comment:
If you go through the `ERROR` condition, widget can only by an `ErrorNotificationWidget`. If you want to declare it with multiple types, declare it before the `ifs`.

You probably did that to avoid declaring it as None, but we shouldn't make the code uglier to fit the typing.

Also, can't widget be declared as a `NotificationWidget` which is the base of all those classes?